[stdlib] Implement Never conformance to Codable

natecook1000 opened the pull request:
This supplies conditional conformance for `Never` to the `Codable` protocol. 

Pitch thread: https://forums.swift.org/t/pitch-conform-never-to-codable/64056/21

natecook1000 committed:
Implement Never conformance to Codable

natecook1000 commented:
@swift-ci Please smoke test

AnthonyLatsis commented on stdlib/public/core/Policy.swift:
```diff
@@ -42,6 +42,23 @@ extension Never: Identifiable {
 
+@available(SwiftStdlib 5.9, *)
+extension Never: Encodable {
+  @available(SwiftStdlib 5.9, *)
```
The extension’s availability should be enough.

natecook1000 committed:
Update for test failures

natecook1000 commented:
@swift-ci Please smoke test

natecook1000 committed:
Add back symbol availability

natecook1000 committed:
Switch to throwing a typeMismatch error

From looking at other failure points, it looks like dataCorrupted
is more used for structural issues, e.g. malformed JSON.

natecook1000 commented on stdlib/public/core/Policy.swift:
```diff
@@ -42,6 +42,23 @@ extension Never: Identifiable {
+@available(SwiftStdlib 5.9, *)
+extension Never: Encodable {
+  @available(SwiftStdlib 5.9, *)
+  public func encode(to: Encoder) throws {}
```
Good catch!

natecook1000 committed:
Use `any` with existentials

natecook1000 commented:
@swift-ci Please test
